### PR TITLE
More autowiring

### PR DESCRIPTION
**Pull request by biozshock**
| Q                                      | A
| -------------------------------------- | ---
| New feature/enhancement? (use the a.x branch)      | [ x ]
| Automated tests included?              | [ x ] 

#### Description:
Continue with autowiring of controllers. Remove `CoreSubscriber` and inject services via `__construct`. Remove `initialize` method.

#### Steps to test this PR:
Probably best is to code review.

1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Check out export works, other parts of the system works.

### CODE REVIEW

**Review by mabumusa1 — DISMISSED** [review]
I tested and it worked as expected, create a couple of contacts, emails, pages, and a report all worked as excepted

File: app/bundles/CoreBundle/Controller/MauticController.php [hunk 1]
```diff
@@ -2,21 +2,9 @@
 
 namespace Mautic\CoreBundle\Controller;
 
-use Mautic\CoreBundle\Factory\ModelFactory;
-use Symfony\Component\HttpKernel\Event\ControllerEvent;
-
 /**
  * A dummy interface to ensure that only Mautic bundles are affected by Mautic onKernelController events.
  */
 interface MauticController
```
Comment (escopecz) [inline app/bundles/CoreBundle/Controller/MauticController.php:8]: I think we can remove this interface completelly as it's not used bu removeing the `onKernelController` method

File: app/bundles/CoreBundle/Controller/Api/FileApiController.php [hunk 2]
```diff
@@ -22,13 +33,13 @@ class FileApiController extends CommonApiController
      */
     protected $allowedExtensions = [];
 
-    public function initialize(ControllerEvent $event)
+    public function __construct(CorePermissions $security, Translator $translator, EntityResultHelper $entityResultHelper, RouterInterface $router, FormFactoryInterface $formFactory, AppVersion $appVersion, RequestStack $requestStack, ManagerRegistry $doctrine, ModelFactory $modelFactory, EventDispatcherInterface $dispatcher, CoreParametersHelper $coreParametersHelper, MauticFactory $factory)
     {
         $this->entityNameOne     = 'file';
         $this->entityNameMulti   = 'files';
         $this->allowedExtensions = $this->coreParametersHelper->get('allowed_extensions');
```
Comment (escopecz) [inline app/bundles/CoreBundle/Controller/Api/FileApiController.php]:
```suggestion
        $this->allowedExtensions = $coreParametersHelper->get('allowed_extensions');
```

**Review by escopecz — CHANGES_REQUESTED** [review]
Some API Library tests are failing on this PR:

https://github.com/mautic/api-library/actions/runs/5141896925/jobs/9254942391

File: app/bundles/CoreBundle/Controller/MauticController.php (hunk 1 above)
Comment (biozshock) [reply app/bundles/CoreBundle/Controller/MauticController.php:8]: I plan on using it later for autowiring purposes. But sure can remove it and add later. WDYT?

**Comment by mabumusa1** [comment]
@biozshock do not worry about failing CS, I think we merged a PR without running CS fixer. I am starting a PR to do that now

File: app/bundles/CoreBundle/Controller/MauticController.php (hunk 1 above)
Comment (escopecz) [reply app/bundles/CoreBundle/Controller/MauticController.php:8]: If you plan to use it then it's OK to leave it there.

**Review by escopecz — APPROVED** [review]
The API Library tests are green: https://github.com/mautic/api-library/actions/runs/5155567039

**Comment by biozshock** [comment]
Though there are probably still php-cs issues... @mabumusa1 please ping me when you are done with the php-cs, or just fix my errors in your PR ;)

**Comment by mabumusa1** [comment]
> Though there are probably still php-cs issues... @mabumusa1 please ping me when you are done with the php-cs, or just fix my errors in your PR ;)

I will fix it, do not worry

**Pull request merged**